connectors/sources/sharepoint_online.py

```diff
@@ -286,7 +287,7 @@ async def wrapped(*args, **kwargs):
             retry = 1
             while retry <= retries:
                 try:
-                    async for item in func(*args, **kwargs):
+                    async for item in func(*args, **kwargs, retry_count=retry):
```

Comment:
Curious why we want to implement a `retryable_aiohttp_call` here instead of using `retryable` in `utils.py`

Reply:
>we don't need to retry when the response code is 404.

You can skip the exception in `skipped_exceptions`:

https://github.com/elastic/connectors-python/blob/6a845a318fb4b80dea23aca917e1631cfe5223b2/connectors/utils.py#L420-L425


> I mentioned in the description that we need to do a refactoring here and maybe extend `retryable` from `utils.py` to support this feature.

Agree. This was discussed before to add throttling support in `retryable`.